Mathlib/RepresentationTheory/Homological/GroupCohomology/Shapiro.lean

```diff
@@ -0,0 +1,63 @@
+/-
+Copyright (c) 2025 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+import Mathlib.CategoryTheory.Preadditive.Projective.Resolution
+import Mathlib.RepresentationTheory.Homological.GroupCohomology.Basic
+import Mathlib.RepresentationTheory.FiniteIndex
+
+/-!
+# Shapiro's lemma for group cohomology
+
+Given a commutative ring `k` and a finite index subgroup `S ≤ G`, the file
+`RepresentationTheory/FiniteIndex.lean` defines a natural isomorphism between the functors
+`Ind_S^G, Coind_S^G : Rep k S ⥤ Rep k G`.
+
+Using this isomorphism, we conclude that the `(Co)ind_S^G` and `Res(S) : Rep k G ⥤ Rep k S` are
+both left and right adjoint to each other, and thus that `Res(S)` is an exact functor which
+preserves projective objects. In particular, given a projective resolution `P` of `k` as a trivial
+`k`-linear `G`-representation, `Res(S)(P)` is a projective resolution of `k` as a trivial
+`k`-linear `S`-representation.
+
+Since `Hom(Res(S)(P), A) ≅ Hom(P, Coind_S^G(A))` for any `S`-representation `A`, we conclude
+Shapiro's lemma for group cohomology: `Hⁿ(G, Coind_S^G(A)) ≅ Hⁿ(S, A)` for all `n`.
+
+## Main definitions
+
+* `groupCohomology.coindIso A n`: Shapiro's lemma for group cohomology: an isomorphism
+  `Hⁿ(G, Coind_S^G(A)) ≅ Hⁿ(S, A)`, given a finite index subgroup `S ≤ G` and an
+  `S`-representation `A`.
+
+!-/
+
+universe u
+
+namespace groupCohomology
+
+open CategoryTheory Finsupp TensorProduct Rep
+
+variable {k G : Type u} [CommRing k] [Group G] {S : Subgroup G}
+  [DecidableRel (QuotientGroup.rightRel S)] [Fintype (G ⧸ S)] (A : Rep k S)
```

Comment:
We have a typeclass for this. You can get the fintype instance from `Subgroup.fintypeQuotientOfFiniteIndex` (a def), and there's also the instance `Subgroup.finite_quotient_of_finiteIndex`.
```suggestion
  [DecidableRel (QuotientGroup.rightRel S)] [S.FiniteIndex] (A : Rep k S)
```

Reply:
(I haven't looked at your code closely, but if you really need a fintype instance (rather than just finite) for the statement of the def, then you could always make `Subgroup.fintypeQuotientOfFiniteIndex` into an instance locally).